Lib/fontTools/feaLib/builder.py

```diff
@@ -704,10 +704,7 @@ def set_language(self, location, language, include_default, required):
                 if key[:2] in self.get_default_language_systems_():
                     lookups = [l for l in lookups if l not in dflt_lookups]
                 self.features_.setdefault(key, []).extend(lookups)
-        if self.script_ == 'DFLT':
-            langsys = set(self.get_default_language_systems_())
```

Comment:
Wait, I’m not 100% sure it’s correct that you got rid of that call to  get_default_language_systems_ here.
Supposedly that was a “feature” (ie. falling back to DFLT dflt when no languagesystems statement are exlicitly defined in feature file). I need to check again again the fea spec. I’ll look at this tomorrow, I’m out of the office now.

Reply:
ok, I finally checked this. Martin is correct, the whole `if` block here was wrong.

The parser asserted that the only valid language tag that could be associated with script "DFLT" was "dflt"; so, what this `if self.script_ == 'DFLT':` block was doing basically was that, if a feature contained explicit `script DFLT; language dflt` statements then any rules or lookup defined after them would be registered not only for DFLT/dflt, but also (and that's incorrect) for all the globally defined languagesystems. And that is why in the PairPosSubtable.ttx expected test result, the same kernlookup was incorrectly registered twice under latn/dflt language system: i.e. once because of the explicit DFLT/dflt that precedes, and again because added specifically to latn/dflt in the following statements.

https://github.com/fonttools/fonttools/blob/632192804fefc4a0f7d37f54e2cc25b82f6b460e/Tests/feaLib/data/PairPosSubtable.fea

What the Feature File spec says, instead, is:

> Rules that are specified after the start of a feature and before the first "script" and/or "language" statement will be included in all the language systems specified by the "languagesystem" statements. If you do not want any of the rules in the feature to be registered under the language systems specified by the "languagesystem" statements, then a "script" and/or "language" statement for a script must be present before the first rule in the feature.

which is exactly what that feature file snippet is doing: i.e. it defines an explicit script/language combination, in this case DFLT/dflt, so the here meaning is: "I do not want to register the following lookup for all the global languagesystems, but only for this specific one".  There is nothing special about DFLT/dflt combination as far as the determination of the _default lookups_. The spec only says that the rules/lookups that occur between the start of the feature and the _first script_ (doesn't say the first non-DFLT script) are are added to all languagesystems.

Thanks @mhosken for spotting this!